fossology/fossology
# feat(newagent/scancode): Integrate ScanCode Toolkit to FOSSology

## Description

Integrating ScanCode toolkit as a new agent to FOSSology. Using the command line interface from the scancode package in order to be called right from the fossology application. Fossology is passing a single file and takes the result for `license/copyright/author` from the scancode command line call.

### Changes

   | category | task | status |
   |----------|------|--------|
   |template|Using scancode custom template| &#9745; |
   |license| license info scanned by scancode | &#9745; |
   |license DB|Reusing existing license in the fossology database| &#9745;|
   |license DB|Inserting new license in the fossology database| &#9745; |
   |license UI|Highlighting license relevance text| &#9745; |
   |license DB|Inserting new license key,full name, ref_url| &#9745; | 
   |license DB|Inserting license text for new license in fossology| &#9744; |
   |license report|Including scancode in the license report| &#9745;|
   |copyright|Creating new tables for copyright and author both on UI and DB| &#9745; |
   |copyright|copyright and author info scanned by scancode| &#9745; |
   |copyright|email and URL info scanned by scancode| &#9744; |
   |copyright DB|Populated newly created scancode database table| &#9745; |
   |copyright UI|copyright bulk view has scancode table populated| &#9745; |
   |copyright UI|Email/URL/Author bulk view has scancode table populated| &#9745; |
   |copyright UI|copyright single view has scancode integrated| &#9745; |
   |copyright UI|Highlight working for copyright| &#9744;|
   |copyright report|Including scancode copyright in the report| &#9744;|
   |scancode|Schedule scancode from bulk view space |&#9744;|
   |API|Integrating ScanCode to FOSSology API |&#9745;|
   |testing|write tests for scancode agent|&#9744;|

## How to test

  - Fetch [sarita/newagent/scancode](https://github.com/itssingh/fossology/tree/sarita/newagent%2Fscancode) on your local by using command `$ git fetch https://github.com/itssingh/fossology sarita/newagent/scancode:scancode` and checkout to the `scancode` branch.
  - [Install FOSSology from source](https://github.com/fossology/fossology/wiki/Install-from-Source)
  - Upload a file from Source/URL/VCS then select license or copyright under scancode-toolkit.

    *NOTE:* Email and URL is not working for now as scancode-toolkit doesn't provide these info by using custom template ([PR](https://github.com/nexB/scancode-toolkit/pull/2539) is merged waiting for new release 🥳 ).

  - Check license browser/license single file view for license information.
  - check copyright bulk view or copyright single view for copyright information.
  - check author/email/url bulk view.

    *NOTE:* Please schedule fossology copyright as well in order to see copyright tables on UI (have to fix this).

- If you are facing problem in installing scancode, please install scancode-toolkit manually. The steps to [install scancode-toolkit](https://github.com/itssingh/fossology/wiki/scancode#the-steps-to-install-scancode-toolkit)



<a href="https://gitpod.io/#https://github.com/fossology/fossology/pull/2074"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
